solstdlib/src/precompiles.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+
+function sha256(bytes memory input) returns (bytes32 ret) {
+  assembly {
+    let success := staticcall(gas(), 2, add(input, 32), mload(input), 0, 32)
```

Comment:
We should consider making this incompatible for pre-tangerineWhistle and use `-1` instead of `gas()` here. Or we query the evm version.

Reply:
> Or we query the evm version.

The idea was to query the version in the step, once we have #10283. It is a necessity imho otherwise it won't be possible to remove the non-stdlib route and still support compiling for older versions.

Nit: `gas()` is cheaper than `push(-1)`.